src/sys/unix/selector/kqueue.rs

```diff
@@ -304,7 +305,30 @@ impl Drop for Selector {
 }
 
 pub type Event = libc::kevent;
-pub type Events = Vec<Event>;
+pub struct Events(Vec<libc::kevent>);
+
+impl Events {
+    pub fn with_capacity(capacity: usize) -> Events {
+        Events(Vec::with_capacity(capacity))
+    }
+}
+
+impl Deref for Events {
+    type Target = Vec<libc::kevent>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for Events {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+unsafe impl Send for Events {}
```

Comment:
Can you add a comment saying why this is safe? For example @asomers comment from https://github.com/nix-rust/nix/pull/442/files#diff-7a42d5ac10efafbf1371c20fe9247662R195-R197.